pkg/tsdb/sql_engine_test.go

```diff
@@ -14,6 +14,28 @@ func TestSqlEngine(t *testing.T) {
 		dt := time.Date(2018, 3, 14, 21, 20, 6, int(527345*time.Microsecond), time.UTC)
 		earlyDt := time.Date(1970, 3, 14, 21, 20, 6, int(527345*time.Microsecond), time.UTC)
 
+		Convey("Given a time range between 2018-04-12 00:00 and 2018-04-12 00:05", func() {
+			from := time.Date(2018, 4, 12, 18, 0, 0, 0, time.UTC)
+			to := from.Add(5 * time.Minute)
+			timeRange := NewFakeTimeRange("5m", "now", to)
+			query := &Query{}
+
+			Convey("interpolate $__interval", func() {
+				sql, err := Interpolate(query, timeRange, "select $__interval ")
+				So(err, ShouldBeNil)
+
+				So(sql, ShouldEqual, "select 300s ")
+			})
+
+			Convey("interpolate $__interval_ms", func() {
```

Comment:
A test where using $__interval in $__timeGroup would be nice since I think that won't work. Think you need to do the global interpolation before the datasource interpolation, but I may be wrong?

Reply:
It should work since the regexp matcher for the datasource looks for function calls and should leave $__interval untouched but actual global macros might not work for that reason and the datasource specific macro handling code would need to be adjusted to let those macros through.
But I'm not 100% sure and will add test cases around this.

I did the calls in this order to allow datasource specific implementation of macros to overwrite the global one.